nixos/modules/services/web-apps/logseq-webapp.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, pkgs, config, utils, ... }: let
+
+  cfg = config.services.logseq-webapp;
+
+in {
+
+  options.services.logseq-webapp = {
+
+    enable = lib.mkEnableOption (lib.mdDoc "Local-first, non-linear, outliner notebook");
+
+    hostname = lib.mkOption {
+      type = lib.types.str;
+      default = "localhost";
+      description = lib.mdDoc ''
+        The hostname on which to listen.
+      '';
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    services.caddy = {
```

Comment:
Do many other services use caddy as their web server? All of the other ones I've seen in nixpkgs have standardized on nginx.

Reply:
There are a few services that use Caddy (e.g. [WordPress](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/web-apps/wordpress.nix#L358) and [DokuWiki](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/web-apps/dokuwiki.nix#L424)), but usually configurable as an alternative to Nginx via the options.